File: Biblioteca/src/main/java/com/biblioteca/domain/material/MaterialDigital.java
package com.biblioteca.domain.material;

import com.biblioteca.util.LoggerUtil;

public final class MaterialDigital extends MaterialBiblioteca{

    static final String URL_BASE_MATERIALES_DIGITALES = "https://test?resource_id=";
    private TipoMaterialDigital tipoMaterialDigital;
    private LoggerUtil log = new LoggerUtil();

    public MaterialDigital(long identificador, TipoMaterialDigital tipoMaterialDigital) {
        super(identificador);
        this.tipoMaterialDigital = tipoMaterialDigital;
    }

    public TipoMaterialDigital getTipoMaterialDigital() {
        return tipoMaterialDigital;
    }

    public void setTipoMaterialDigital(TipoMaterialDigital tipoMaterialDigital) {
        this.tipoMaterialDigital = tipoMaterialDigital;
    }

    public void getResource() {
        log.info("OBTENIENDO RECURSO A TRAVÉS DEL ENLACE: " + URL_BASE_MATERIALES_DIGITALES + this.getIdentificador());
    }


}

File: python/biblioteca/src/domain/material/material_digital.py
from .material_biblioteca import MaterialBiblioteca
from .tipo_material_digital import TipoMaterialDigital

class MaterialDigital(MaterialBiblioteca):
    URL_BASE_MATERIALES_DIGITALES = 'https://test?resource_id='

    def __init__(self, identificador, tipo_material_digital = TipoMaterialDigital):
        super().__init__(identificador)
        self.tipo_material_digital = tipo_material_digital
    
    def __str__(self):
        return f"""
        MaterialBiblioteca(
        identificador: {self.identificador},
        tipo_material_digital: {self.tipo_material_digital})
        """
    
    def get_resource(self):
        return f"{self.URL_BASE_MATERIALES_DIGITALES}{self.identificador}"
    